provider/internal/queue/provide_test.go

```diff
@@ -0,0 +1,287 @@
+package queue
+
+import (
+	"crypto/rand"
+	"testing"
+
+	mh "github.com/multiformats/go-multihash"
+
+	"github.com/probe-lab/go-libdht/kad/key/bitstr"
+	"github.com/probe-lab/go-libdht/kad/trie"
+	"github.com/stretchr/testify/require"
+
+	"github.com/libp2p/go-libp2p-kad-dht/provider/internal/keyspace"
+)
+
+func genMultihashesMatchingPrefix(prefix bitstr.Key, n int) []mh.Multihash {
+	mhs := make([]mh.Multihash, 0, n)
+	for i := 0; len(mhs) < n; i++ {
+		digest := make([]byte, 32)
+		if _, err := rand.Read(digest); err != nil {
+			panic(err)
+		}
+		h, err := mh.Encode(digest, mh.SHA2_256)
+		if err != nil {
+			panic(err)
+		}
```

Comment:
Look at [`random.Multihashes`](https://pkg.go.dev/github.com/ipfs/go-test@v0.2.2/random#Multihashes) or even [`random.Bytes`](https://pkg.go.dev/github.com/ipfs/go-test@v0.2.2/random#Bytes)
```suggestion
		h := random.Multihashes(1)[0]
```